src/Microsoft.Health.Dicom.Core/Features/Store/Entries/DicomInstanceEntryReaderForSinglePartRequest.cs

```diff
@@ -52,10 +62,110 @@ public async Task<IReadOnlyList<IDicomInstanceEntry>> ReadAsync(string contentTy
                     string.Format(CultureInfo.InvariantCulture, DicomCoreResource.UnsupportedContentType, contentType));
             }
 
-            Stream seekableStream = await _seekableStreamConverter.ConvertAsync(stream, cancellationToken);
+            Stream seekableStream;
```

Comment:
nit: there should be a comment as to why you can dispose of the underlying stream so early. It's not very obvious